### PR TITLE
DROOLS-2304: [DMN Designer] Add context entry from literal expression cell

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-2304

### CODE REVIEW

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/literal/LiteralExpressionGridTest.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        final GridData mockParentUiModel = mock(GridData.class);
+        final BaseExpressionGrid mockParentGrid = mock(BaseExpressionGrid.class);
+        doReturn(mockParentUiModel).when(parent).getGridData();
+        doReturn(mockParentUiModel).when(mockParentGrid).getModel();
+        doReturn(new HashSet<GridWidget>() {{
+            add(mockParentGrid);
+            add(grid);
+        }}).when(gridLayer).getGridWidgets();
+
+        grid.fireExpressionEditorSelectedEvent();
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/test/java/org/kie/workbench/common/dmn/client/editors/expressions/types/literal/LiteralExpressionGridTest.java]: I think we could use `grid.select();` at this place to cover more the reported problem.

File: kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/literal/LiteralExpressionGrid.java [hunk 2]
```diff
@@ -109,4 +110,21 @@ protected void initialiseUiModel() {
     public Optional<IsElement> getEditorControls() {
         return Optional.empty();
     }
+
+    @Override
+    protected void fireExpressionEditorSelectedEvent() {
+        final Optional<BaseExpressionGrid> parentGrid = findParentGrid(parent);
```
Comment (jomarko) [inline kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/types/literal/LiteralExpressionGrid.java]: Maybe your version is more readable, just wanted to ask if you noticed.
Here you could update to:
```java
final Optional<BaseExpressionGrid> parentGrid = findParentGrid();
```
if the `findParentGrid` declared as:
```java
private Optional<BaseExpressionGrid> findParentGrid()
```

**Comment by manstis** [comment]
@jomarko Thanks for your review; updates made as suggested in the 2nd commit.

**Pull request merged**